wagtail/wagtailcore/middleware.py

```diff
@@ -1,15 +1,9 @@
 from __future__ import absolute_import, unicode_literals
 
-import django
+from django.utils.deprecation import MiddlewareMixin
```

Comment:
This is a Django <1.10 compatibility shim https://docs.djangoproject.com/en/1.11/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware and should probably be eliminated too now.

Reply:
class-based middleware is still usable in Django 1.11 though

Reply:
Yep, that's true - it should continue to work for as long as `django.utils.deprecation.MiddlewareMixin` exists, and there's no indication that Django intends to remove it any time soon. It's more of a nice-to-have, in the interests of eliminating legacy code and fully following Django >=1.10 conventions.